### PR TITLE
[backend] test injects execution (#1486)

### PR DESCRIPTION
**Pull request by johanah29**
#1486

### CODE REVIEW

File: openbas-api/src/test/java/io/openbas/rest/InjectApiTest.java [hunk 1]
```diff
@@ -339,6 +370,142 @@ void updateInjectForSimulationTest() throws Exception {
     assertEquals(injectTitle, JsonPath.read(response, "$.inject_title"));
   }
 
+  @DisplayName("Execute an email inject for exercise")
+  @Test
+  @WithMockPlannerUser
+  void executeEmailInjectForExerciseTest() throws Exception {
```
Comment (MarineLeM) [inline openbas-api/src/test/java/io/openbas/rest/InjectApiTest.java:376]: to ensure a complete test, is it possible to verify the recipient and the subject of the email that has been sent?

File: openbas-api/src/test/resources/application.properties [hunk 2]
```diff
@@ -3,55 +3,47 @@
...
 #############
 # INJECTORS #
 #############
-
 # Mail config (Always available)
-openbas.default-mailer=no-reply@openbas.io
 openbas.default-reply-to=contact@openbas.io
-spring.mail.host=smtp.gmail.com
-spring.mail.port=587
+openbas.default-mailer=no-reply@openbas.io
```
[the remark this answers is not in the document]
Comment (johanah29) [reply openbas-api/src/test/resources/application.properties:37]: File was aligned with master branch

File: openbas-api/src/test/resources/application.properties
Comment (guillaumejparis) [inline openbas-api/src/test/resources/application.properties:1]: todo: revert this file
as @MarineLeM said you need to revert for clarity there are carriage return on master :
https://github.com/OpenBAS-Platform/openbas/blob/master/openbas-api/src/test/resources/application.properties

File: openbas-api/src/test/java/io/openbas/utils/fixtures/ExerciseFixture.java [hunk 3]
```diff
@@ -44,4 +44,13 @@ public static Exercise createDefaultIncidentResponseExercise() {
...
     return exercise;
   }
+
+  /*
+  _ create exercise with team containing a user with email (user enabled)
+  _ create exercise with team containing a user with email (user not enabled)
+  _ create exercise without team
+  _ create exercise with team without user
+  _ create email
+   */
```
Comment (guillaumejparis) [inline openbas-api/src/test/java/io/openbas/utils/fixtures/ExerciseFixture.java:54]: question: is this useful ?

File: openbas-api/src/test/java/io/openbas/rest/InjectApiTest.java [hunk 4]
```diff
@@ -339,6 +371,145 @@ void updateInjectForSimulationTest() throws Exception {
...
+
+  @DisplayName("Execute an email inject for exercise with no content")
+  @Test
+  @WithMockPlannerUser
+  void executeEmailInjectForExerciseWithNoContentTest() throws Exception {
+    // -- PREPARE --
+    InjectorContract injectorContract =
+        this.injectorContractRepository.findById(EMAIL_DEFAULT).orElseThrow();
+    Inject inject = getInjectForEmailContract(injectorContract);
+    User user = userRepository.findById(currentUser().getId()).orElseThrow();
```
Comment (MarineLeM) [inline openbas-api/src/test/java/io/openbas/rest/InjectApiTest.java:485]: you can remove this statement, user is not use in this test

File: openbas-api/src/test/java/io/openbas/rest/InjectApiTest.java [hunk 5]
```diff
@@ -339,6 +371,145 @@ void updateInjectForSimulationTest() throws Exception {
...
+
+  @DisplayName("Execute an email inject for exercise with no team")
+  @Test
+  @WithMockPlannerUser
+  void executeEmailInjectForExerciseWithNoTeam() throws Exception {
+    // -- PREPARE --
+    InjectorContract injectorContract =
+        this.injectorContractRepository.findById(EMAIL_DEFAULT).orElseThrow();
+    Inject inject = getInjectForEmailContract(injectorContract);
+    User user = userRepository.findById(currentUser().getId()).orElseThrow();
```
Comment (MarineLeM) [inline openbas-api/src/test/java/io/openbas/rest/InjectApiTest.java:443]: you can remove this statement, user is not use in this test

**Review by MarineLeM — APPROVED** [review]

**Pull request merged**